prime-router/src/main/kotlin/common/HttpClientUtils.kt

```diff
@@ -530,43 +553,32 @@ class HttpClientUtils {
          * note: most configuration parameters are overridable
          * e.g. expectSuccess default to false because most of the time
          * the caller wants to handle the whole range of response status
-         * @param bearerTokens null default, the access token needed to call the endpoint
+         * @param bearerTokens the access token needed to call the endpoint
          * @return a HttpClient with all sensible defaults
          */
-        fun createDefaultHttpClient(accessToken: String?): HttpClient {
-            return HttpClient(Apache) {
-                // installs logging into the call to post to the server
-                // commented out - not to override underlying default logger settings
-                // enable to trace http client internals when needed
-                // install(Logging) {
-                //     logger = Logger.SIMPLE
-                //     level = LogLevel.INFO
-                // }
-                // not using Bearer Auth handler due to refresh token behavior
-                accessToken?.let {
-                    defaultRequest {
-                        header("Authorization", "Bearer $it")
+        fun getDefaultHttpClient(): HttpClient {
+            synchronized(this) {
```

Comment:
This relates to my comment about having httpClient initialized at the creation of the object instead of here and make it not resettable: we should evaluate if we really need this if we apply the suggest httpClient change.

Reply:
ack